dcicutils/lang_utils.py

```diff
@@ -0,0 +1,156 @@
+from .misc_utils import ignored
+import re
+
+
+class EnglishUtils:
+    """
+    In most cases you can get away without using this class, but it's here in case customization is needed.
+    The usual interfaces are available as separate functions.
+    """
+
+    SECOND = 1
+    MINUTE = 60 * SECOND
+    HOUR = 60 * MINUTE
+    DAY = 24 * HOUR
+    WEEK = 7 * DAY
+
+    _TIME_UNITS = ((WEEK, "week"), (DAY, "day"), (HOUR, "hour"), (MINUTE, "minute"))
```

Comment:
It's linear because the algorithm relies on iterating in order.
It's a tuple rather than a list because originally I used a recursive algorithm that wanted it to be a default argument, and lists are a no-no as a default argument. Under some circumstances, and this is one, a tuple is a non-modifiable list. Alas, Python is irregular about that and should really have a way to make a read-only list.